File: cloud/acp-admin-cloud-dependencies/src/main/java/pers/acp/admin/common/permission/oauth/AuthConfigExpression.java
package pers.acp.admin.common.permission.oauth;

import pers.acp.admin.common.constant.ModuleFuncCode;
import pers.acp.admin.common.constant.RoleCode;
import pers.acp.admin.common.permission.BaseExpression;

/**
 * 定义权限配置权限表达式
 *
 * @author zhang by 28/12/2018
 * @since JDK 11
 */
public interface AuthConfigExpression extends BaseExpression {

    /**
     * 权限配置
     */
    String authConfig = "hasAnyAuthority('" + RoleCode.prefix + RoleCode.ADMIN + "','" + ModuleFuncCode.authConfig + "')";

    /**
     * 权限新增
     */
    String authAdd = "hasAnyAuthority('" + RoleCode.prefix + RoleCode.ADMIN + "','" + ModuleFuncCode.authAdd + "')";

    /**
     * 权限删除
     */
    String authDelete = "hasAnyAuthority('" + RoleCode.prefix + RoleCode.ADMIN + "','" + ModuleFuncCode.authDelete + "')";

    /**
     * 权限更新
     */
    String authUpdate = "hasAnyAuthority('" + RoleCode.prefix + RoleCode.ADMIN + "','" + ModuleFuncCode.authUpdate + "')";

    /**
     * 权限查询
     */
    String authQuery = "hasAnyAuthority('" + RoleCode.prefix + RoleCode.ADMIN + "','" + ModuleFuncCode.authQuery + "')";

}

File: cloud/oauth-server/src/main/kotlin/pers/acp/admin/oauth/constant/AuthConfigExpression.kt
package pers.acp.admin.oauth.constant

import pers.acp.admin.constant.ModuleFuncCode
import pers.acp.admin.constant.RoleCode

/**
 * 定义权限配置权限表达式
 *
 * @author zhang by 28/12/2018
 * @since JDK 11
 */
object AuthConfigExpression {
    /**
     * 权限配置
     */
    const val authConfig =
        "hasAnyAuthority('" + RoleCode.prefix + RoleCode.SUPER + "','" + ModuleFuncCode.authConfig + "')"

    /**
     * 权限新增
     */
    const val authAdd = "hasAnyAuthority('" + RoleCode.prefix + RoleCode.SUPER + "','" + ModuleFuncCode.authAdd + "')"

    /**
     * 权限删除
     */
    const val authDelete =
        "hasAnyAuthority('" + RoleCode.prefix + RoleCode.SUPER + "','" + ModuleFuncCode.authDelete + "')"

    /**
     * 权限更新
     */
    const val authUpdate =
        "hasAnyAuthority('" + RoleCode.prefix + RoleCode.SUPER + "','" + ModuleFuncCode.authUpdate + "')"

    /**
     * 权限查询
     */
    const val authQuery =
        "hasAnyAuthority('" + RoleCode.prefix + RoleCode.SUPER + "','" + ModuleFuncCode.authQuery + "')"
}